xds/src/main/java/io/grpc/xds/Filter.java

```diff
@@ -103,6 +119,36 @@ default ServerInterceptor buildServerInterceptor(
     return null;
   }
 
+  /**
+   * Implement to perform cleanup on filter shutdown.
```

Comment:
This isn't javadoc style for the first line. Maybe: "Releases filter resources like shared resources and remote connections"?